File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.*;
import java.util.*;

public class Main {
    static int n; // 모든 컴퓨터의 수
    static List<ArrayList<Integer>> network = new ArrayList<>();
    static boolean[] visited;
    static int count = 0;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        n = Integer.parseInt(br.readLine()); // 모든 컴퓨터의 수 초기화
        int edge = Integer.parseInt(br.readLine()); // 네트워크 상에서 직접 연결되어 있는 컴퓨터 쌍의 수

        for (int i = 0; i <= n; i++) {
            network.add(new ArrayList<>());
        }

        for (int i = 0; i < edge; i++) { // 주어진 간선 리스트로 인접 리스트 구성
            StringTokenizer st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            // 양방향 처리
            network.get(a).add(b);
            network.get(b).add(a);
        }
        visited = new boolean[n+1]; // 방문 횟수 처리 초기화

        dfs(1);
        bw.write(count + " ");
        bw.flush();
        bw.close();
    }

    static void dfs(int start) {
        visited[start] = true;
        for (int next : network.get(start)) {
            if (!visited[next]) {
                dfs(next);
                count++;
            }

        }
    }
}

File: 백준/Silver/2606. 바이러스/바이러스.py
import sys

def dfs(graph, visited, node):
    visited[node] = True
    count = 1  # 현재 노드 감염

    for neighbor in graph[node]:
        if not visited[neighbor]:  # 방문하지 않은 경우
            count += dfs(graph, visited, neighbor)

    return count

# 입력 처리
n = int(sys.stdin.readline())  # 컴퓨터 수
m = int(sys.stdin.readline())  # 연결된 간선 수
graph = {i: [] for i in range(1, n + 1)}  # 그래프 초기화

for _ in range(m):
    a, b = map(int, sys.stdin.readline().split())
    graph[a].append(b)
    graph[b].append(a)  # 양방향 연결

visited = [False] * (n + 1)

infected_count = dfs(graph, visited, 1) - 1  # 1번 컴퓨터 제외
print(infected_count)